src/app/(public)/[username]/[slug]/page.tsx

```diff
@@ -14,29 +14,27 @@ export async function generateMetadata(
 ): Promise<Metadata> {
   const article = await getArticleById(getIdFromSlug(props.params.slug));
   if (!article) {
-    throw new Error("Article not found");
+    return {};
   }
 
   const imageUrl = article.cover || "";
-  const author = article.author;
 
-  if (!author) {
-    return {};
-  }
   return {
     title: article.title,
     description: article.description,
-    creator: author.username,
+    creator: article.author.username,
     authors: [
       {
-        name: author.username,
-        url: `${process.env.SITE_URL}/${author.username}` || "",
+        name: article.author.username,
+        url: `${process.env.SITE_URL}/${article.author.username}` || "",
       },
     ],
     publisher: "KhmerCoders",
     openGraph: {
       type: "article",
-      url: `${process.env.SITE_URL}/${author.username}/${article.slug}` || "",
+      url:
+        `${process.env.SITE_URL}/${article.author.username}/${article.slug}` ||
```

Comment:
Env should be added to the configuration

Reply:
i added to env example

Reply:
`@/lib/env.mjs`

<img width="1802" alt="image" src="https://github.com/invisal/khmercoders/assets/106462074/253b52fd-9d5b-4593-b1f0-031955386d94">

1. Add into the `client` object since this is used client side
2. Make sure the name is prepend with `NEXT_PUBLIC` e.g `NEXT_PUBLIC_SITE_URL`
3. To use it `import { env } from '@/lib/env.mjs'` then `env.NEXT_PUBLIC_SITE_URL`